### DISCUSSION TITLE
[BP#17] Choisir un format de données adapté

### DISCUSSION THREAD

**Discussion by Altheachevalier** [cnumr/best-practices#92] [Bonnes pratiques] [answered]
Il est primordial de connaitre ses données et également le format de stockage. 
Si une entreprise détient des données stockées dans une base et que toutes celles-ci sont stockées en « chaine de caractères » (pour rappel cela permet d’être permissif sur les données), elle va avoir beaucoup plus de bit de stockage par rapport à un stockage en nombre entier (per exemple interger).

La chaine de caractère prend environ 250bits de stockage contre environ 5 bits de stockage pour un nombre entier.  SI on multiplie cela par 10 000 lignes de stockage dans un data center, cela aura un impact énorme sur les coûts d’infrastructures mais également sur l’électricité engendrée par le stockage physique.

**Comment by ldesgrange** [comment]
N'est-ce pas identique à la [BP17](https://github.com/cnumr/best-practices/blob/main/chapters/BP_017_fr.md)?

**Reply by devcrafting** [reply]
Je n'ai pas compris non plus 🤔

**Comment by Altheachevalier** [comment]
Effectivement, il serait intéressant de rajouter ces éléments à cette BP,
Merci beaucoup

**Reply by ACTLEM** [reply]
J'ai modifié cette discussion pour taguer en "modification".

**Comment by fzaninotto** [comment]
> SI on multiplie cela par 10 000 lignes de stockage dans un data center, cela aura un impact énorme sur les coûts d’infrastructures mais également sur l’électricité engendrée par le stockage physique.

Je suis curieux de voir l'étude scientifique qui sous-tend ce raisonnement. Parce que mon intuition va dans l'autre sens : ça aura un impact négligeable.

**Comment by ACTLEM** [comment]
@Altheachevalier Est-il possible de préciser la modification que tu vois dans cette bonne pratique stp?

**Reply by Altheachevalier** [reply]
Il serait simplement intéressant de parler des formats de stockage dans la BP 17 : notamment le stockage en nombre entier vs en chaine de caractère afin de minimiser le poids de stockage.

**Reply by ACTLEM** [reply]
Ce n'est pas déjà le cas dans l'exemple?

**Reply by Altheachevalier** [reply]
D'après ma compréhension la BP n'évoque pas les différences de chaque format de stockage et leur poids. Il serait intéressant de les notifier pour évoquer les différents choix avec les poids de stockage, non?

**Reply by ACTLEM** [reply]
On pourrait effectivement ajouter un exemple (ou modifier l'exemple) en mettant en avant les formats selon le type (chaîne de caractère, nombre, objet binaire...) et pas seulement sur un type identique.

**Reply by Altheachevalier** [reply]
On peut modifier l'exemple en indiquant à minima les différents types de stockage ?

**Reply by devcrafting** [reply]
Est-ce que ce n'est pas trop spécifique au SQL ?
Dans l'exemple que je donnais sur la BP https://github.com/cnumr/best-practices/discussions/116#discussioncomment-1610759 la dénormalisation était dans une table id / JSON, de quoi en faire bondir plus d'un et pourtant si on avait dû recréer un modèle relationnel optimisé...on aurait perdu tout l'avantage de la dénormalisation. Donc je suis assez nuancé sur cette BP.

**Reply by Altheachevalier** [reply]
Je ne pense pas que cela soit spécifique au SQL, il s'agit simplement de notifier les types de stockage  et leur poids peut importe si cela est du SQL ou non. 
L'objectif ici de plus de sensibiliser à la connaissance de sa donnée et également de son type de stockage pour minimiser le poids final. 
Qu'en pensez-vous?

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé l'issue pour ajouter des informations sur les types de stockage/poids: https://github.com/cnumr/best-practices/issues/257
